pkg/telemetry/phonehome/request_params.go

```diff
@@ -57,6 +58,34 @@ func (rp *RequestParams) HasUserAgentWith(substrings []string) bool {
 	return false
 }
 
+func hasValueMatching(values []string, expression string) bool {
+	for _, value := range values {
+		if ok, _ := regexp.MatchString(expression, value); ok {
```

Comment:
For reviewers' attention: header regular expressions is a short list of strings.
Shall they be compiled and cached?